### PR TITLE
nixos/security/pam: added `origin` option to pamu2f

### PR DESCRIPTION
**Pull request by shimunn**
###### Description of changes

Added the `origin` option to `security.pam.u2f`

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by arianvp — APPROVED** [review]

File: nixos/modules/security/pam.nix [hunk 1]
```diff
@@ -878,6 +878,24 @@ in
         '';
       };
 
+      origin = mkOption {
+        default = null;
+        type = with types; nullOr str;
+        description = ''
+            By default <literal>pam-u2f</literal> module sets the origin
+            to <literal>pam://$HOSTNAME</literal>.
+            Setting origin to an host independet value will allow you to
```
Comment (lostnet) [inline nixos/modules/security/pam.nix:887]: typo s/independet/independent

**Review by lostnet — APPROVED** [review]

**Review by lostnet — APPROVED** [review]

**Review by LunNova — APPROVED** [review]

**Comment by shimunn** [comment]
@arianvp any further changes required?

File: nixos/modules/security/pam.nix [hunk 2]
```diff
@@ -470,7 +470,7 @@ let
             auth ${p11.control} ${pkgs.pam_p11}/lib/security/pam_p11.so ${pkgs.opensc}/lib/opensc-pkcs11.so
           '') +
           (let u2f = config.security.pam.u2f; in optionalString cfg.u2fAuth ''
-            auth ${u2f.control} ${pkgs.pam_u2f}/lib/security/pam_u2f.so ${optionalString u2f.debug "debug"} ${optionalString (u2f.authFile != null) "authfile=${u2f.authFile}"} ${optionalString u2f.interactive "interactive"} ${optionalString u2f.cue "cue"} ${optionalString (u2f.appId != null) "appid=${u2f.appId}"}
+            auth ${u2f.control} ${pkgs.pam_u2f}/lib/security/pam_u2f.so ${optionalString u2f.debug "debug"} ${optionalString (u2f.authFile != null) "authfile=${u2f.authFile}"} ${optionalString u2f.interactive "interactive"} ${optionalString u2f.cue "cue"} ${optionalString (u2f.appId != null) "appid=${u2f.appId}"} ${optionalString (u2f.origin != null) "origin=${u2f.origin}"}
```
Comment (SuperSandro2000) [inline nixos/modules/security/pam.nix:473]: That line is getting way to long. We should split it up a bit.

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by arianvp** [comment]
If you could squash all the commits into one so that we don't have commits that don't follow the contribution guidelines, I think we're all good and I'll merge!

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by stigtsp** [comment]
This change might break pam with the following config:
```
  security.pam = {
    services.sudo.u2fAuth = true;
    u2f.cue = true;
  };

```

```
Jul 21 11:39:31 host su[7329]: PAM (su) illegal module type: cue
Jul 21 11:39:31 host su[7329]: PAM (su) no control flag supplied
Jul 21 11:39:31 host su[7329]: PAM (su) no module name supplied
```

```
root@host /e/pam.d# grep cue *
chfn: cue  
chpasswd: cue  
chsh: cue  
cups: cue  
groupadd: cue  
groupdel: cue  
groupmems: cue  
groupmod: cue  
i3lock: cue  
i3lock-color: cue  
login: cue  
passwd: cue  
polkit-1: cue  
runuser: cue  
runuser-l: cue  
sshd: cue  
su: cue  
sudo:auth sufficient /nix/store/fdvbhzdvl4i8hr4vvk0nry701sj869nw-pam_u2f-1.2.0/lib/security/pam_u2f.so    cue  
systemd-user: cue  
useradd: cue  
userdel: cue  
usermod: cue  
vlock: cue  
xlock: cue  
xscreensaver: cue  


```

**Pull request merged**